### ISSUE TITLE
requestedPixelSize = 0 fails deadly

### ISSUE THREAD

**Issue by SalmaDammak** [qupath/qupath#777]
**Describe the bug**
I erroneously set requestedPixelSize = 0 instead of downsample = 0 in my exporting script that I adapted from [the docs](https://qupath.readthedocs.io/en/0.2/docs/advanced/exporting_annotations.html), the script didn't error but instead, after about a day running and my RAM filling up to about 20GB I realized my mistake. I suspect requestedPixelSize = 0 causes an infinite loop somewhere or the divide by zero on lines 378 and 379 of TileExporter are handled strangely.

**To Reproduce**
In the script from [the docs](https://qupath.readthedocs.io/en/0.2/docs/advanced/exporting_annotations.html) set equestedPixelSize = 0 

**Expected behavior**
It would be nice to have some input validation or a way to raise an exception for this use case. 

**Desktop (please complete the following information):**
OS: Windows 7
QuPath Version 0.2.3

**Comment by petebankhead** [comment]
Oh dear... could you try [v0.3.0-rc1](https://github.com/qupath/qupath/releases) to see if it has been fixed? There have been some fixes in that part of the code that may help, but I'm not sure.

If that doesn't resolve things, could you please post the script that's failing? There are several on the linked page and so I'm not sure which exactly is going wrong and where.

**Comment by petebankhead** [comment]
I'll close this issue because, as far as I can tell, it has already been fixed in v0.3.0-rc1 and in the main branch.

I see the error
```
IllegalArgumentException at line 17: Downsample must be > 0, but this one is 0.0
```
if I run the following script
```groovy
import qupath.lib.images.servers.LabeledImageServer

def imageData = getCurrentImageData()

// Define output path (relative to project)
def name = GeneralTools.getNameWithoutExtension(imageData.getServer().getMetadata().getName())
def pathOutput = buildFilePath(PROJECT_BASE_DIR, 'export', name)
mkdirs(pathOutput)

// Define output resolution
double requestedPixelSize = 0

// Convert to downsample
double downsample = requestedPixelSize / imageData.getServer().getPixelCalibration().getAveragedPixelSize()

// Create an ImageServer where the pixels are derived from annotations
def labelServer = new LabeledImageServer.Builder(imageData)
    .backgroundLabel(0, ColorTools.WHITE) // Specify background label (usually 0 or 255)
    .downsample(downsample)    // Choose server resolution; this should match the resolution at which tiles are exported
    .addLabel('Tumor', 1)      // Choose output labels (the order matters!)
    .addLabel('Stroma', 2)
    .addLabel('Other', 3)
    .lineThickness(2)          // Optionally export annotation boundaries with another label
    .setBoundaryLabel('Boundary*', 4) // Define annotation boundary label
    .multichannelOutput(false) // If true, each label refers to the channel of a multichannel binary image (required for multiclass probability)
    .build()


// Export each region
int i = 0
for (annotation in getAnnotationObjects()) {
    def region = RegionRequest.createInstance(
        labelServer.getPath(), downsample, annotation.getROI())
    i++
    def outputPath = buildFilePath(pathOutput, 'Region ' + i + '.ome.tif')
    writeImageRegion(labelServer, region, outputPath)
}
```
